src/main/java/seedu/storage/ContactsDecoder.java

```diff
@@ -0,0 +1,35 @@
+package seedu.storage;
+
+import seedu.contact.Contact;
+import seedu.contact.ContactList;
+import seedu.contact.DetailType;
+import seedu.exception.FileErrorException;
+import static seedu.storage.Storage.SEPARATOR;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class ContactsDecoder {
```

Comment:
I like the use of ContactsDecoder and ContactsEncoder to abstract the code, makes it a lot more readable.